pilot/pkg/model/service.go

```diff
@@ -609,18 +610,45 @@ func ParseServiceKey(s string) (hostname Hostname, ports PortList, labels Labels
 	return
 }
 
+// defaultSubsetName - Use this name for subset is not defined.
+const defaultSubsetName = "_default_"
+
 // BuildSubsetKey generates a unique string referencing service instances for a given service name, a subset and a port.
 // The proxy queries Pilot with this key to obtain the list of instances in a subset.
 func BuildSubsetKey(direction TrafficDirection, subsetName string, hostname Hostname, port *Port) string {
-	return fmt.Sprintf("%s|%s|%s|%s", direction, port.Name, subsetName, hostname)
+	if subsetName == "" {
+		subsetName = defaultSubsetName
+	}
+
+	portName := port.Name
+	if portName == "" {
+		portName = strconv.FormatInt(int64(port.Port), 10)
+	}
+
+	return fmt.Sprintf("%s.%s.%s.%s", direction, portName, subsetName, hostname)
```

Comment:
if direction == outbound - skip direction.
If inbound: it's probably ok to start with "in" ( keep in mind there is a limit in cluster names, full inbound doesn't help)

Reply:
We are well within the limit. I don’t mind changing inbound to in if you drop support for eds in Envoy v1. Otherwise you have no way of distinguishing the class of rewquestd